executers/dependenciesutils.go

```diff
@@ -59,6 +73,30 @@ func collectDependenciesAndPublish(targetRepo string, failOnError, publishDeps b
 	return nil
 }
```

Comment:
I'd like to suggest the following change to the above code.
With this change, the  ```if publishDeps``` question is asked only once (instead of twice.
I also split and simplified the description a bit.
Let me know what you think:
```
	// If we need to publish the missing depedencies to Artifactory
	if publishDeps {
		// If we need publish the missing depedencies to an Artifactory server or repo, which are
		// different then the resolution repo, then we have no information about which depedencies are actually
		// missing and therefore should be published.
		// Let's find out which depedencies are missing.
		if !reflect.DeepEqual(resolverDeployer.Resolver(), resolverDeployer.Deployer()) {
			err = prepareDependenciesForDeployerServer(&cache, dependenciesToPublish, resolverDeployer)
			if err != nil {
				return err
			}
		}
		// Publish the missing depedencies to Artifactory.
		err = populateAndPublish(resolverDeployer.Deployer().Repo(), cachePath, dependenciesInterface, packageDependencies, &cache, resolverDeployer.Deployer().ServiceManager())
		if err != nil {
			return err
		}
	}

	utils.LogFinishedMsg(&cache)
	return nil

Reply:
Done